NitroxModel/Packets/InitialPlayerSync.cs

```diff
@@ -26,7 +26,7 @@ public class InitialPlayerSync : Packet
         public bool FirstTimeConnecting { get; set; }
         public InitialPDAData PDAData { get; set; }
         public InitialStoryGoalData StoryGoalData { get; set; }
-        public HashSet<string> CompletedGoals { get; set; }
```

Comment:
HashSet<T> was not being serialized correctly due to having private internal variables. I am working on updating it to allow explicit support of private fields using attributes, but this will not work because the type is not defined in a Nitrox assembly.

Reply:
I don't believe there is any performance benefit of using a HashSet here because the initial sync processor simply adds the values to another HashSet. However, the property type should be changed to ICollection<string> because the serializer will then be able to take advantage of knowing the length of the collection ahead of time.